app_legacy/Helpers/render/set-claim.php

```diff
@@ -74,25 +72,47 @@ function renderFinishedClaimsComponent(int $count): void
     echo "</tr>";
     echo "</thead>";
     echo "<tbody>";
-    foreach ($claimData as $claim) {
-        $claimUser = $claim['User'];
-        echo "<tr>";
-        echo "<td class='pr-0'>";
-        echo userAvatar($claimUser, label: false);
-        echo "</td>";
-        echo "<td>";
-        echo userAvatar($claimUser, icon: false);
-        echo "</td>";
-        echo "<td class='pr-0'>";
-        echo gameAvatar($claim, label: false);
-        echo "</td>";
-        echo "<td class='w-full'>";
-        echo gameAvatar($claim, icon: false);
-        echo "</td>";
-        echo "<td>" . ($claim['SetType'] == ClaimSetType::NewSet ? ClaimSetType::toString(ClaimSetType::NewSet) : ClaimSetType::toString(ClaimSetType::Revision)) . "</td>";
-        echo "<td class='smalldate'>" . getNiceDate(strtotime($claim['DoneTime'])) . "</td>";
-        echo "</tr>";
-    }
+
+    $remaining = $count;
+    $offset = 0;
+    do {
```

Comment:
Wow, I just realized that I haven't seen a `do {} while()` loop in a while :)
Regardless, I assume this this can be simplified in the future to not pass the offset and count as soon as the changes of https://github.com/RetroAchievements/RAWeb/pull/1393 are in?

Reply:
I don't think #1393 will have any impact on this. The problem being addressed is that the query is returning items where the console ID has not yet been validated (console is rollout, claim was completed in unofficial, so players can't actually see or play it yet). This solution queries the first $count items, and then applies the filter locally. If any items were filtered, it asks for me.

I realize it's not the most efficient way to fetch the additional records, but there's no easy way to turn `isValidConsoleId` into a CSV for use in the query, and the second fetch is going to be unnecessary most of the time.

Reply:
wait, that was in a different PR.
`getFilteredClaims()` already allows to retrieve all entries by not passing the `$limit` parameter?

Reply:
i understand now 👍